### PR TITLE
ENT-4655 |unlink enterprise learners command

### PR DESCRIPTION
**Pull request by HammadAhmadWaqas**
**Merge checklist:**
- [ ] Any new requirements are in the right place (do **not** manually modify the `requirements/*.txt` files)
    - `base.in` if needed in production but edx-platform doesn't install it
    - `test-master.in` if edx-platform pins it, with a matching version
    - `make upgrade && make requirements` have been run to regenerate requirements
- [ ] `make static` has been run to update webpack bundling if any static content was updated
- [ ] `./manage.py makemigrations` has been run
    - Checkout the [Database Migration](https://openedx.atlassian.net/wiki/spaces/AC/pages/23003228/Everything+About+Database+Migrations) Confluence page for helpful tips on creating migrations.
    - *Note*: This **must** be run if you modified any models.
      - It may or may not make a migration depending on exactly what you modified, but it should still be run.
    - This should be run from either a venv with all the lms/edx-enterprise requirements installed or if you checked out edx-enterprise into the src directory used by lms, you can run this command through an lms shell.
        - It would be `./manage.py lms makemigrations` in the shell.
- [ ] [Version](https://github.com/edx/edx-enterprise/blob/master/enterprise/__init__.py) bumped
- [ ] [Changelog](https://github.com/edx/edx-enterprise/blob/master/CHANGELOG.rst) record added
- [ ] Translations updated (see docs/internationalization.rst but also this isn't blocking for merge atm)

**Post merge:**
- [ ] Tag pushed and a new [version](https://github.com/edx/edx-enterprise/releases) released
    - *Note*: Assets will be added automatically. You just need to provide a tag (should match your version number) and title and description.
- [ ] After versioned build finishes in [GitHub Actions](https://github.com/edx/edx-enterprise/actions), verify version has been pushed to [PyPI](https://pypi.org/project/edx-enterprise/)
    - Each step in the release build has a condition flag that checks if the rest of the steps are done and if so will deploy to PyPi.
    (so basically once your build finishes, after maybe a minute you should see the new version in PyPi automatically (on refresh))
- [ ] PR created in [edx-platform](https://github.com/edx/edx-platform) to upgrade dependencies (including edx-enterprise)
    - This **must** be done after the version is visible in PyPi as `make upgrade` in edx-platform will look for the latest version in PyPi.
    - Note: the edx-enterprise constraint in edx-platform **must** also be bumped to the latest version in PyPi.

### CODE REVIEW

File: enterprise/management/commands/unlink_enterprise_customer_learners.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    def handle(self, *args, **options):
+        csv_path = options['data_csv']
+        enterprise_customer_uuid = options['enterprise_customer_uuid']
+        enterprise_customer = EnterpriseCustomer.objects.get(uuid=enterprise_customer_uuid)
+
+        LOGGER.info('Started Unlinking and removing DSC')
+
+        results = {}
+
+        with open(csv_path) as csv_file:
```
Comment (georgebabey) [inline enterprise/management/commands/unlink_enterprise_customer_learners.py:53]: For a one-time thing this is probably fine, but I would recommend reading the file first and then doing the operations outside of the `with open` so you don't keep the file handler open longer than you need to.

File: enterprise/management/commands/unlink_enterprise_customer_learners.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+                        email=email, ec_name=enterprise_customer.name
+                    )
+                    results[email]['Unlinked'] = message
+                    continue
+
+                user = User.objects.get(email=email)
+                enterprise_customer_user = get_enterprise_customer_user(user.id, enterprise_customer_uuid)
+                data_sharing_consent_records = enterprise_customer_user.data_sharing_consent_records
+
+                # Deleting the DCS cache
```
Comment (georgebabey) [inline enterprise/management/commands/unlink_enterprise_customer_learners.py:74]: But: DSC not DCS

File: enterprise/management/commands/unlink_enterprise_customer_learners.py [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+from enterprise.utils import get_enterprise_customer_user
+
+User = auth.get_user_model()  # pylint: disable=invalid-name
+
+LOGGER = logging.getLogger(__name__)
+
+
+class Command(BaseCommand):
+    """
+    Django management command for creating DataSharingConsent records
```
Comment (sameenfatima78) [inline enterprise/management/commands/unlink_enterprise_customer_learners.py:24]: Nit: Incorrect docstring

File: enterprise/management/commands/unlink_enterprise_customer_learners.py [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+                    results[email]['Unlinked'] = message
+                    continue
+
+                user = User.objects.get(email=email)
+                enterprise_customer_user = get_enterprise_customer_user(user.id, enterprise_customer_uuid)
+                data_sharing_consent_records = enterprise_customer_user.data_sharing_consent_records
+
+                # Deleting the DCS cache
+                deleted_course_ids = []
+                for data_sharing_consent in data_sharing_consent_records:
```
Comment (georgebabey) [inline enterprise/management/commands/unlink_enterprise_customer_learners.py:76]: We also need to delete the actual DSC record, not just the cache (although the cache was a great catch I had forgotten about)

File: enterprise/management/commands/unlink_enterprise_customer_learners.py [hunk 5]
```diff
@@ -0,0 +1,86 @@
+# -*- coding: utf-8 -*-
+"""
+Django management command for unlink the learners and removing DSC.
```
Comment (sameenfatima78) [inline enterprise/management/commands/unlink_enterprise_customer_learners.py:3]: nit: for unlinking

**Review by georgebabey — CHANGES_REQUESTED** [review]

File: enterprise/management/commands/unlink_enterprise_customer_learners.py [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+            dest='enterprise_customer_uuid',
+            help='Run this command for only the given EnterpriseCustomer UUID.'
+        )
+
+    def handle(self, *args, **options):
+        csv_path = options['data_csv']
+        enterprise_customer_uuid = options['enterprise_customer_uuid']
+        enterprise_customer = EnterpriseCustomer.objects.get(uuid=enterprise_customer_uuid)
+
+        LOGGER.info('Started Unlinking and removing DSC')
```
Comment (zamanafzal) [inline enterprise/management/commands/unlink_enterprise_customer_learners.py:49]: Can we please add logs with a constant primary string like 
LOGGER.info('[Unlink and Remove DSC] Started Unlinking and removing DSC')

File: enterprise/management/commands/unlink_enterprise_customer_learners.py (hunk 3 above)
Comment (zamanafzal) [inline enterprise/management/commands/unlink_enterprise_customer_learners.py:24]: nit: Please update the docstring.

File: enterprise/management/commands/unlink_enterprise_customer_learners.py [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+    """
+    Django management command for creating DataSharingConsent records
+    """
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            '--data-csv',
+            action='store',
+            dest='data_csv',
+            help='Path of csv to read enterprise learner data',
```
Comment (zamanafzal) [inline enterprise/management/commands/unlink_enterprise_customer_learners.py:32]: nit: Path of csv to read enterprise learner email.

**Review by georgebabey — APPROVED** [review]

**Review by sameenfatima78 — APPROVED** [review]

**Pull request merged**